common/pkg/libartifact/artifact.go

```diff
@@ -58,9 +59,22 @@ type ArtifactList []*Artifact
 // GetByNameOrDigest returns an artifact, if present, by a given name
```

Comment:
Unrelated and _should not_ be fixed in this PR, just something I happened to notice: `GetDigest` re-creates a JSON from parsed data, that’s not guaranteed to compute the right digest.